account_invoice_import_invoice2data/models/invoice2data_template.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import base64
+import cgi
+import json
+import logging
+import os
+import pkg_resources
+import pprint
+import sys
+
+from openerp import _, api, fields, models
+from openerp.tools import file_open, float_compare
+from tempfile import mkstemp
+
+try:
+    from invoice2data.in_pdftotext import to_text
+    from invoice2data.main import extract_data
+    from invoice2data.main import logger as loggeri2data
+    from invoice2data.template import InvoiceTemplate
+    from invoice2data.template import read_templates
+    from invoice2data.utils import ordered_load
+except ImportError:
+    logging.error('Failed to import invoice2data')
+
+
+def _fork_stdin(stdin, func):
+    """fork and call func in the child process, writing stdin to child's fd"""
+    read_stdin, write_stdin = os.pipe()
+    read_fd, write_fd = os.pipe()
+    child_pid = os.fork()
+    if not child_pid:
+        sys.stdin.close()
+        os.dup2(read_stdin, 0)
+        os.close(write_stdin)
+        os.close(read_fd)
+        with os.fdopen(write_fd, 'w') as f:
+            f.write(func())
+    else:
+        os.close(write_fd)
+        with os.fdopen(write_stdin, 'w') as f:
+            f.write(stdin)
+        os.waitpid(child_pid, 0)
+        with os.fdopen(read_fd, 'r') as f:
+            return f.read()
+
+
+class Invoice2dataTemplate(models.Model):
+    _name = 'invoice2data.template'
+    _description = 'Template for invoice2data'
+
+    name = fields.Char(required=True)
+    template_type = fields.Selection([('pi', 'Purchase Invoice')], 'Type',
```

Comment:
I really dislike this 'pi' abbreviation. If you need to make other changes anyway and want to do me a favour, replace it by 'purchase_invoice'.